### PR TITLE
Add logical model entity to FSH grammar

### PR DESCRIPTION
**Pull request by mint-thompson**
Completes task [CIMPL-687](https://standardhealthrecord.atlassian.net/browse/CIMPL-687).

For details about logical models, visit the official FHIR documentation: http://hl7.org/fhir/R4/structuredefinition.html.

A logical model is a special kind of StructureDefinition that follows some additional constraints and allows authors to construct more or less arbitrary data structures within FHIR. As such, any FSH rule that can appear on a StructureDefinition can also appear on a logical model. A logical model definition may also contain addElementRules, which allow authors to add new elements to the logical model. This rule incorporates cardinality, flags, and types, allowing for concise definitions in many cases.

There are no test cases associated with this PR, since the importer is not changed. Only the grammar definition and parser contexts are changed as part of this PR.

### CODE REVIEW

File: antlr/src/main/antlr/FSH.g4 [hunk 1]
```diff
@@ -3,14 +3,17 @@ grammar FSH;
 options { tokenVocab = FSHLexer; }
 
 doc:                entity* EOF;
-entity:             alias | profile | extension | invariant | instance | valueSet | codeSystem | ruleSet | paramRuleSet | mapping;
+entity:             alias | profile | extension | invariant | instance | valueSet | codeSystem | ruleSet | paramRuleSet | mapping | logical;
```
Comment (cmoesel) [inline antlr/src/main/antlr/FSH.g4:6]: It looks like this needs `resource` too:
```suggestion
entity:             alias | profile | extension | invariant | instance | valueSet | codeSystem | ruleSet | paramRuleSet | mapping | logical | resource;
```

File: src/import/parserContexts.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ export interface EntityContext extends ParserRuleContext {
   ruleSet(): RuleSetContext;
   paramRuleSet(): ParamRuleSetContext;
   mapping(): MappingContext;
+  logical(): LogicalContext;
```
Comment (cmoesel) [inline src/import/parserContexts.ts:18]: We need `resource()` here too:
```suggestion
  logical(): LogicalContext;
  resource(): ResourceContext;
```

**Review by cmoesel — CHANGES_REQUESTED** [review]
Looks good -- I think there are just a could spots that need `resource` added.

**Review by julianxcarter — DISMISSED** [review]
Aside from adding the resource keyword in the spots that Chris pointed out, this looks great to me! Approving preemptively 👍🏽

File: antlr/src/main/antlr/FSH.g4 [hunk 3]
```diff
@@ -3,14 +3,17 @@ grammar FSH;
...
 doc:                entity* EOF;
-entity:             alias | profile | extension | invariant | instance | valueSet | codeSystem | ruleSet | paramRuleSet | mapping;
+entity:             alias | profile | extension | invariant | instance | valueSet | codeSystem | ruleSet | paramRuleSet | mapping | logical | resource;
 
 alias:              KW_ALIAS SEQUENCE EQUAL SEQUENCE;
 
 profile:            KW_PROFILE name sdMetadata+ sdRule*;
 extension:          KW_EXTENSION name sdMetadata* sdRule*;
+logical:            KW_LOGICAL name sdMetadata* lrRule*;
+resource:           KW_RESOURCE name sdMetadata lrRule*;
```
Comment (cmoesel) [inline antlr/src/main/antlr/FSH.g4:13]: Oops.  Missed this one in my last review.
```suggestion
resource:           KW_RESOURCE name sdMetadata* lrRule*;
```

File: antlr/src/main/antlr/FSH.g4 [hunk 4]
```diff
@@ -3,14 +3,17 @@ grammar FSH;
 options { tokenVocab = FSHLexer; }
 
 doc:                entity* EOF;
-entity:             alias | profile | extension | invariant | instance | valueSet | codeSystem | ruleSet | paramRuleSet | mapping;
+entity:             alias | profile | extension | invariant | instance | valueSet | codeSystem | ruleSet | paramRuleSet | mapping | logical | resource;
```
Comment (cmoesel) [inline antlr/src/main/antlr/FSH.g4:6]: b146f0c added `resource` but doesn't show any of the changes I'd expect in the ANTLR-generated files.  I think we need to re-generate those files.

**Review by cmoesel — CHANGES_REQUESTED** [review]
Ever so close.  Just one minor tweak and a re-run of the ANTLR code generator is needed.

**Review by cmoesel — APPROVED** [review]
This looks good!

**Comment by cmoesel** [comment]
And @julianxcarter's pre-emptive approval is good enough to justify a merge.

**Pull request merged**